packages/formik/src/Formik.tsx

```diff
@@ -152,6 +152,7 @@ export function useFormik<Values extends FormikValues = FormikValues>({
   const initialTouched = React.useRef(props.initialTouched || emptyTouched);
   const initialStatus = React.useRef(props.initialStatus);
   const isMounted = React.useRef<boolean>(false);
+  const validated = React.useRef<boolean>(false);
```

Comment:
I think your move to add a ref here is right, but we should be running validation synchronously on mount if `validateOnMount` is set such that validation has occurred within the first render pass and there is no potential for the boolean to flip either way.

Reply:
@quantizor  any hint ? this cause annoying UI issue for me

Reply:
Any chance to fix it ? I really need this fix